ui/desktop/app/routes/scopes/scope/projects/targets/target.js

```diff
@@ -42,22 +42,24 @@ export default class ScopesScopeProjectsTargetsTargetRoute extends Route {
        * from the model hook for hosts
        */
       try {
-        const hostSets = await Promise.all(
-          target.host_sources.map(({ host_source_id }) =>
-            this.store.findRecord('host-set', host_source_id),
-          ),
-        );
+        const { host_sources } = target;
+        let allFilteredHosts = [];
 
-        // Extract host ids from all host sets
-        const hostIds = hostSets.flatMap(({ host_ids }) => host_ids);
+        for (const hostSource of host_sources) {
+          const query = {
+            host_catalog_id: hostSource.host_catalog_id,
+          };
+          const hostsQuery = await this.store.query('host', query);
 
-        // Load unique hosts
-        const uniqueHostIds = new Set(hostIds);
+          allFilteredHosts.push(
+            ...hostsQuery.filter((host) =>
+              host.host_set_ids?.includes(hostSource.host_source_id),
+            ),
+          );
+        }
 
-        hosts = await Promise.all(
-          [...uniqueHostIds].map((hostId) =>
-            this.store.findRecord('host', hostId),
-          ),
+        hosts = Array.from(
+          new Map(allFilteredHosts.map((host) => [host.id, host])).values(),
```

Comment:
What is the purpose of this?

Reply:
Remove duplicate hosts. I tested this by creating 2 host-sets in a host-catalog then creating 3 hosts. I added host 1 and 2 to the first host-set and host 2 and 3 to the second host-set. This resulted in duplicate hosts showing up to the user.

Reply:
Ah, I see I didn't think about duplicate hosts, can we add a comment explaining that then?